### DISCUSSION TITLE
Colors in Final phase of the competition

### DISCUSSION THREAD

**Discussion by MJavadZallaghi** [cyberbotics/wrestling#88] [Q&A] [answered]
@omichel Hi; Hope you are doing well.
In the [results page](https://github.com/cyberbotics/wrestling/blob/main/result.md), I saw that the colors of all competitors are defined in the final phase of the competition. Previously, it was mentioned that the colors will be chosen randomly for each match, as you have stated this [here](https://github.com/cyberbotics/wrestling/discussions/38). Please let me know if I'm wrong.

I've a suggestion: Consider we have a match between X, and Y. The colors can be assigned as below, and this way is more justly.

<table>
  <tr>
    <th>Colors</th>
    <th>X->Red, Y->Blue</th>
    <th>X->Blue, Y->Red</th>
    <th>X->random, Y->random</th>
  </tr>
  <tr>
    <td>Game</td>
    <td>1</td>
    <td>2</td>
    <td>3</td>
  </tr>
</table>

Also maybe it can be a better idea if the games could held in 3 win from 5 game format. In this way, the effect of chance can be reduced more in comparison to 2 win from 3 game. In that case, the colors can be defined as follow:

<table>
  <tr>
    <th>Colors</th>
    <th>X->Red, Y->Blue</th>
    <th>X->Blue, Y->Red</th>
    <th>X->Red, Y->Blue</th>
    <th>X->Blue, Y->Red</th>
    <th>X->random, Y->random</th>
  </tr>
  <tr>
    <td>Game</td>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td>
    <td>5</td>
  </tr>
</table>

**Comment by omichel** [comment, marked as answer]
I said the color would be chosen randomly because at this time, we didn't yet defined the exact algorithm for the finals. In any case, it will be somehow random, depending on the ranking of the participants. The details were announced [two weeks ago](https://github.com/cyberbotics/wrestling/discussions/63#discussioncomment-5882678) and nobody reacted to this, so I assumed it way OK for everyone.

Anyway, thank you for your proposals, they are very good indeed. However, I don't want to change the rules in the very last hours before the finals as some other competitors may find this last minute change unfair. If you would have suggested this earlier, we could have changed these rules, but now it is too late. Nevertheless, I will take this into account for the next edition of the competition.

**Comment by MJavadZallaghi** [comment]
Okay, and thank you for all the efforts from the start of the competition,

Best regards,
Mohammad Javad.
